docs/native-modules-setup.md

```diff
@@ -45,154 +45,64 @@ The steps to create a new native module library project are:
-1. Select `Add` > `Existing Project...`.
-1. Select the project file and click `Open`.
+Take the result of that command (let's say it's `16.13.1`) and use it to upgrade the dev dependency:
 
-When you are done, your solution should look like this:
-![native module dependencies](assets/native-module-dependencies.png)
+```bat
+yarn upgrade react@16.13.1 --dev
+```
 
-You now have all of the React Native Windows projects to your solution. Next we're going to reference them in our `MyLibrary` project.
+You'll need to repeat the steps for `react-native`, ie:
 
-### Referencing React Native Windows in your Project
+```bat
+npm info react-native-windows@canary devDependencies.react-native
+```
 
-The only project reference you **must** add is `Microsoft.ReactNative`. To add the reference:
+Again, take the result of that command (let's say it's `0.0.0-a36d9cd7e`) and use it to upgrade the dev dependency:
 
-1. Open the Solution Explorer sidebar.
-1. Right-click on your `MyLibrary` project.
-1. Select `Add` > `Reference`.
-1. Select `Projects` on the left-hand side.
-1. Check the box next to `Microsoft.ReactNative`.
-1. Click `OK`.
+```bat
+yarn upgrade react-native@0.0.0-a36d9cd7e --dev
+```
 
-After you've added the reference, you need to make sure it doesn't copy itself into your build (otherwise it'll cause build conflicts down the line when you're trying on consume your library):
+Now you should be ready to add Windows support with the CLI.
 
-1. Open the Solution Explorer sidebar.
-1. Under your `MyLibrary` project, expand the `References`.
-1. Right-click on `Microsoft.ReactNative`.
-1. Select `Properties`.
-1. Under `Build`, Change `Copy Local` to `False`.
+## Adding Windows support to an existing library
 
-Now, you're technically ready to go, but in order to improve the developer experience, it's also **highly recommended** to also add a reference to the appropriate helper shared project. These projects contain the attributes (C#) and macros (C++) as described in the [Native Modules](native-modules.md) and [View Managers](view-managers.md) documents.
+Similar to adding Windows support to an app project, you can use the CLI to add Windows support to a native module project, but you'll need to specify `--projectType lib`:
 
-If you're writing in C#, you'll want to add `Microsoft.ReactNative.SharedManaged`:
+```bat
+npx react-native-windows-init --version canary --projectType lib --overwrite
+```
 
-1. Open the Solution Explorer sidebar.
-1. Right-click on your `MyLibrary` project.
-1. Select `Add` > `Reference`.
-1. Select `Shared Projects` on the left-hand side.
-1. Check the box next to `Microsoft.ReactNative.SharedManaged`.
-1. Click `OK`.
+This defaults to a C++/WinRT project. If you want to create a C# based native module project, use:
 
-If you're writing in C++, you'll want to add `Microsoft.ReactNative.Cxx`:
+```bat
+npx react-native-windows-init --version canary --projectType lib --language cs --overwrite
```

Comment:
What does `react-native-windows-init` lib mode do for dependencies/peer dependencies? Feels like we might need to do something manually here. I.e. we don't want any deps on react-native-windows, but do want a permissive peer dependency.

Reply:
Th logic is here: https://github.com/microsoft/react-native-windows/blob/master/packages/react-native-windows-init/src/Cli.ts#L331

From what I can see, only apps depend directly on react, react-native, and/or react-native-windows. Libraries may have zero dependencies, and at minimum just have react-native as a dev dependency.

So if I see dependencies with react-native (probably an app), we use the previous logic of inserting rnw into dependencies and running yarn/npm install.

If I see 'react-native' as a dev dependency (probably a lib), then add rnw as a dev dependency with yarn/npm

Reply:
Makes sense. I think we need to add a step asking to add a peer dependency then, since the CLI just adds a dev dependency.

Reply:
I think there needs to some "global" statement that if you're adding windows implementation to a community module, you'll need to work with the module owners when making any `package.json` changes.